tests/AbstractFixerTest.php

```diff
@@ -89,9 +89,14 @@ public function testGetWhitespacesFixerConfig(): void
 
         $fixer->setWhitespacesConfig($newConfig);
 
-        $config = $fixer->getWhitespacesConfig();
+        $config = self::getWhitespaceConfig($fixer);
 
         self::assertSame("\t", $config->getIndent());
         self::assertSame("\r\n", $config->getLineEnding());
     }
+
+    private static function getWhitespaceConfig(AbstractFixer $fixer): WhitespacesFixerConfig
+    {
+        return \Closure::bind(static fn (AbstractFixer $fixer): WhitespacesFixerConfig => $fixer->whitespacesConfig, null, $fixer)($fixer);
```

Comment:
I think I found out why I don't like the "doMagic" or those Closure binds.

because we are hacky and test internals of the method, that we shall not have access to.

Overall, i believe we should come up to better approach of testing than grabbing values of internal properties/methods (But we already do it on master, I'n OK to keep doing in for scope of this PR)

------

side now, sth we do already in this project in few places can lead us to following:
```
--- a/tests/AbstractFixerTest.php
+++ b/tests/AbstractFixerTest.php
@@ -14,6 +14,7 @@
 
 namespace PhpCsFixer\Tests;
 
+use PhpCsFixer\AccessibleObject\AccessibleObject;
 use PhpCsFixer\AbstractFixer;
 use PhpCsFixer\Tests\Double\FixerDoubleFactory;
 use PhpCsFixer\WhitespacesFixerConfig;
@@ -79,8 +80,9 @@ public function testSetWhitespacesConfigUnconfigurable(): void
     public function testGetWhitespacesFixerConfig(): void
     {
         $fixer = FixerDoubleFactory::createWhitespacesAwareFixer();
+        $fixer = new AccessibleObject($fixer);
 
-        $config = self::getWhitespaceConfig($fixer);
+        $config = $fixer->whitespacesConfig;
 
         self::assertSame('    ', $config->getIndent());
         self::assertSame("\n", $config->getLineEnding());
@@ -89,14 +91,9 @@ public function testGetWhitespacesFixerConfig(): void
 
         $fixer->setWhitespacesConfig($newConfig);
 
-        $config = self::getWhitespaceConfig($fixer);
+        $config = $fixer->whitespacesConfig;
 
         self::assertSame("\t", $config->getIndent());
         self::assertSame("\r\n", $config->getLineEnding());
     }
-
-    private static function getWhitespaceConfig(AbstractFixer $fixer): WhitespacesFixerConfig
-    {
-        return \Closure::bind(static fn (AbstractFixer $fixer): WhitespacesFixerConfig => $fixer->whitespacesConfig, null, $fixer)($fixer);
-    }
 }
```